src/main/scala/rocket/Multiplier.scala

```diff
@@ -42,23 +43,23 @@ class MulDiv(cfg: MulDivParams, width: Int, nXpr: Int = 32) extends Module {
   private def minMulLatency = (cfg.mulUnroll > 0).option(if (cfg.mulEarlyOut) 2 else w/cfg.mulUnroll)
   def minLatency: Int = (minDivLatency ++ minMulLatency).min
 
-  val io = new MultiplierIO(width, log2Up(nXpr))
+  val io = IO(new MultiplierIO(width, log2Up(nXpr)))
   val w = io.req.bits.in1.getWidth
   val mulw = if (cfg.mulUnroll == 0) w else (w + cfg.mulUnroll - 1) / cfg.mulUnroll * cfg.mulUnroll
   val fastMulW = if (cfg.mulUnroll == 0) false else w/2 > cfg.mulUnroll && w % (2*cfg.mulUnroll) == 0
  
-  val s_ready :: s_neg_inputs :: s_mul :: s_div :: s_dummy :: s_neg_output :: s_done_mul :: s_done_div :: Nil = Enum(UInt(), 8)
-  val state = Reg(init=s_ready)
+  val s_ready :: s_neg_inputs :: s_mul :: s_div :: s_dummy :: s_neg_output :: s_done_mul :: s_done_div :: Nil = Enum(8)
+  val state = RegInit(s_ready)
  
-  val req = Reg(io.req.bits)
-  val count = Reg(UInt(width = log2Ceil(
+  val req = Reg(chiselTypeOf(io.req.bits))
```

Comment:
@aswaterman this is the sort of thing I always get nervous about. The old `Reg(io.req.bits)` was just taking the type of that parameter, correct? It's not like a `RegNext` or `RegInit`?

Reply:
You got it right.